src/vic3/effect_validation.rs

```diff
@@ -243,6 +244,57 @@ pub fn validate_country_value(
     vd.field_script_value("value", sc);
 }
 
+pub fn validate_create_building(
+    _key: &Token,
+    block: &Block,
+    _data: &Everything,
+    sc: &mut ScopeContext,
+    mut vd: Validator,
+    _tooltipped: Tooltipped,
+) {
+    vd.req_field("building");
+    vd.field_item("building", Item::BuildingType);
+    let building = block.get_field_value("building");
+    vd.field_validated_list("activate_production_methods", |token, data| {
+        data.verify_exists(Item::ProductionMethod, token);
+        if let Some(building) = building {
+            if let Some((_, block, building_item)) =
+                data.get_item::<BuildingType>(Item::BuildingType, building.as_str())
+            {
+                building_item.validate_production_method(token, building, block, data);
+            }
+        }
+    });
+    vd.field_bool("subsidized");
+    vd.field_numeric_range("reserves", 0.0..=1.0);
+    vd.field_validated_value("level", |_, mut vd| {
+        vd.maybe_is("arable_land");
+        vd.integer();
+    });
+    vd.field_validated_block("add_ownership", |block, data| {
+        let mut vd = Validator::new(block, data);
+        vd.multi_field_validated_block("country", |block, data| {
+            let mut vd = Validator::new(block, data);
+            vd.req_field("country");
+            vd.field_target("country", sc, Scopes::Country);
+            vd.req_field("levels");
+            vd.field_script_value("levels", sc);
```

Comment:
might be hardcoded `field_integer`

Reply:
Yeah I don't know so I went with the less restrictive one. I guess I could test it

Reply:
I've tested this in my mod, seems that both `level` in `create_building` and `levels` in `add_ownership` inside `create_building` accept only `field_integer`, while `script value`, `value scope` and `variable` will all generate an error.

P.S. Actually this is very annoying, so I even create a `scripted effect` to simulate `create_building` while accepting any kinds of `level` by enuming the `level` and expand it to `create_building` call with `field_integer`

Reply:
Many thanks for testing it :)